.github/workflows/validation.yaml

```diff
@@ -0,0 +1,52 @@
+name: Validation
+on:
+  push:
+    branches:
+      - '*'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref || github.run_id }}
```

Comment:
When is the `ref` undefined?

Reply:
This is a very good question, it is copy and paste from Tamanu workflow and I didn't look into that deep. 

`github.ref` would be the branch name for Tupaia in most cases, unless someone uses `git tag` and leave `undefined` message, `github.ref` could be undefined here.